strongbox-security/strongbox-authentication-registry/src/main/java/org/carlspring/strongbox/authentication/DatabaseExternalUsersCacheManager.java

```diff
@@ -26,28 +28,68 @@
 @Transactional
 public class DatabaseExternalUsersCacheManager extends DatabaseUserService implements StrongboxExternalUsersCacheManager
 {
+    
+    private static final Logger logger = LoggerFactory.getLogger(DatabaseExternalUsersCacheManager.class);
 
 
+    @Override
+    public UserEntity findByUsername(String username)
+    {
+        UserEntity result = super.findByUsername(username);
+        if (result != null)
+        {
+            logger.debug("User found in DB cache: username=[{}], sourceId=[{}], id=[{}], uuid=[{}]",
+                         result.getUsername(),
+                         result.getSourceId(),
+                         result.getNativeId(),
+                         result.getUuid());
+        }
+        else
+        {
+            logger.debug("User not found in DB cache: username=[{}]", username);
+        }
+        
+        return result;
+    }
+
     @Override
     @CacheEvict(cacheNames = CacheName.User.AUTHENTICATIONS, key = "#p1.username")
     public User cacheExternalUserDetails(String sourceId,
                                          UserDetails springUser)
     {
-        User user = springUser instanceof StrongboxUserDetails ? ((StrongboxUserDetails) springUser).getUser()
-                : new UserData(springUser);
+        User user;
+        if (springUser instanceof StrongboxUserDetails)
+        {
+            user = ((StrongboxUserDetails) springUser).getUser();
+        }
+        else
+        {
+            user = new UserData(springUser);
+        }
         String username = user.getUsername();
+        logger.info("Cache external user: username=[{}], id=[{}], uuid=[{}], sourceId=[{}], UserDetails=[{}]",
+                    username,
+                    user.getNativeId(),
+                    user.getUuid(),
+                    sourceId,
+                    springUser.getClass().getSimpleName());
         
-        Optional<UserEntity> optionalUser = Optional.ofNullable(findByUsername(user.getUsername()));
+        Optional<UserEntity> oldUser = Optional.ofNullable(findByUsername(user.getUsername()));
         try
         {
             // If found user was from another source then remove before save
-            if (optionalUser.map(User::getSourceId).map(sourceId::equals).filter(Boolean.FALSE::equals).isPresent())
+            Optional<String> oldSource = oldUser.map(User::getSourceId);
+            if (oldSource.map(sourceId::equals).filter(Boolean.FALSE::equals).isPresent())
             {
-                deleteByUsername(optionalUser.map(u -> user.getUuid()).get());
-                optionalUser = Optional.empty();
+                logger.info("Invalidate user from another source: username=[{}], oldSource=[{}], newSource=[{}]",
```

Comment:
This should probablt be `debug` instead.

Reply:
fixed